### ISSUE TITLE
Add badges feature

### ISSUE THREAD

**Issue by willemarcel** [hotosm/tasking-manager#1617]
The users should be rewarded with badges by they contributions. Probably we will need to add a new Badge model and some logic on the User model to calculate the badges a User has conquered.

**Comment by JorgeMartinezG** [comment]
@willemarcel do you know where I can find the list of badges and the number of contributions required for each?

**Comment by willemarcel** [comment]
@JorgeMartinezG I think it needs to be defined yet.
cc @xamanu @smit1678

**Comment by smit1678** [comment]
@willemarcel @xamanu @JorgeMartinezG Yes, I think some of this ties back to the conversations had at the stakeholders meeting around Stats and Metrics and then also to the desire to link to the existing badge model and tracking that happens through the Missing Maps Leaderboards and API: http://www.missingmaps.org/leaderboards/#/missingmaps

**Comment by Aadesh-Baral** [comment]
Like in the missing maps user profiles page add badges on My contributions page based on user contributions on TM and OSM. Badges are also present on TM4 design so UI design of where to display it is already available. We can take references of [missing maps badges logic](https://github.com/MissingMaps/users/tree/master/src/badge_logic) for the implementation. I think the addition of badges will help in user retention in TM.
![user_profile_tm4design](https://user-images.githubusercontent.com/67958673/181170474-b3f143bc-0c76-4127-97f6-24444413ec21.png)

Possible TM badges:

Mapper [Level 1 to 5 based on no. of tasks mapped]
Validator [Level 1 to 5 based on no.of tasks validated]
Project Manager [Level 1 to 5 based on no. of projects successfully managed i.e completed 100%]
Traveller [Level 1 to 3 based on countries mapped]
Work for cause [Level 1 to 3 based on no. of causes contributed to]
Invincible [If the mapped task is never invalidated, needs at least 100 tasks to be mapped, will be revoked if any of the mapped tasks is invalidated,]
Let's add more to this list that will be interesting to the users, help in project completion, and assure data quality. Let's make some badges harder to achieve and maintain so that the user actually needs to put the effort in order to maintain it
